components/object_store/src/obkv/mod.rs

```diff
@@ -0,0 +1,913 @@
+            .await
+            .box_err()
+            .context(ReadMeta {
+                path: location.as_ref().to_string(),
+            })?;
+
+        if let Some(m) = meta {
+            Ok(m)
+        } else {
+            MetaNotExists {
+                path: location.as_ref().to_string(),
+            }
+            .fail()
+        }
+    }
+
+    async fn get_internal(&self, location: &Path) -> std::result::Result<GetResult, Error> {
+        let meta = self.read_meta(location).await?;
+        let table_name = self.pick_shard_table(location);
+        let mut futures = FuturesOrdered::new();
+        for path in meta.parts {
+            let client = self.client.clone();
+            let table_name = table_name.to_string();
+            let future = async move {
+                match client.get(&table_name, path.as_bytes()) {
+                    Ok(res) => Ok(Bytes::from(res.unwrap())),
+                    Err(err) => Err(StoreError::Generic {
+                        store: OBKV,
+                        source: Box::new(err),
+                    }),
+                }
+            };
+            futures.push_back(future);
+        }
+
+        let boxed = futures.boxed();
+
+        Ok(GetResult::Stream(boxed))
+    }
+}
+
+#[async_trait]
+impl<T: TableKv> ObjectStore for ObkvObjectStore<T> {
+    async fn put(&self, location: &Path, bytes: Bytes) -> Result<()> {
+        let instant = Instant::now();
+        // Check if the size of bytes is too large.
+        self.check_size(&bytes)
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        // Use `put_multipart` to implement `put`.
+        let (_upload_id, mut multipart) = self.put_multipart(location).await?;
+        multipart
+            .write(&bytes)
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+        // Complete stage: flush buffer data to obkv, and save meta data
+        multipart
+            .shutdown()
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+        debug!(
+            "ObkvObjectStore put operation, location:{},cost:{}",
+            location,
+            instant.elapsed().as_millis()
+        );
+        Ok(())
+    }
+
+    async fn put_multipart(
+        &self,
+        location: &Path,
+    ) -> Result<(MultipartId, Box<dyn AsyncWrite + Unpin + Send>)> {
+        let instant = Instant::now();
+
+        let upload_id = self.upload_id.fetch_add(1, Ordering::SeqCst);
+        let multi_part_id = format!("{upload_id}");
+        let table_name = self.pick_shard_table(location);
+
+        let upload = ObkvMultiPartUpload {
+            location: location.clone(),
+            upload_id: multi_part_id.clone(),
```

Comment:
Why to declare it as string? Shall we keep it as a number?

Reply:
Because the returned `MultipartId` of method `put_multipart` is a String.